lib/r10k/source/branchname.rb

```diff
@@ -0,0 +1,112 @@
+require 'r10k/git'
+require 'r10k/environment'
+require 'r10k/util/purgeable'
+require 'r10k/util/core_ext/hash_ext'
+
+# This class controls the behaviour of branch names
+#
+# A branch name becomes the name of the environment on the file system.
+# The default branch name is the name of the source branch.
+# The name can be modified with a prefix, and by default will be automatically
+# modified if the branch name contains characters invalid for a file system.
+#
+class R10K::Source::BranchName
+
+  # @!attribute [r] name
+  #   @return [String] The raw name of the branch directly from the source
+  attr_reader :name
+
+  # @!attribute [r] remote
+  #   @return [String] The url of the svn remote. Only used to storage and
+  #     retrieval when working with this class.
+  attr_reader :remote
```

Comment:
Yea, when I first started refactoring I had the exact same thought process.  I nuked it, then it broke the svn class. 

I think I mentioned that in my comment above.   It could be made into a more generic hash, or SVN could handle things better and not need to push anything into BranchName just for storage. 

I let it go on first round of refactoring, not sure if SVN will be end of life at some point? : )